core/node/eth_sender/src/aggregator.rs

```diff
@@ -402,11 +404,10 @@ impl Aggregator {
         limit: usize,
         last_sealed_l1_batch: L1BatchNumber,
         priority_tree_start_index: Option<usize>,
+        execution_delay: Duration,
     ) -> Result<Option<ExecuteBatches>, EthSenderError> {
-        let max_l1_batch_timestamp_millis = self
-            .config
-            .l1_batch_min_age_before_execute
-            .map(|age| unix_timestamp_ms() - age.as_millis() as u64);
+        let max_l1_batch_timestamp_millis =
+            Some(unix_timestamp_ms() - execution_delay.as_millis() as u64);
```

Comment:
Please add some safety margin. On L1 time is discrete and in worst case if you send time at `X`, it will be included in the block with timestamp `X - 12`. So the margin should be greater than 12 sec, I think 30 seconds would ok